File: pirplePython/fizzbuzzprime.py
"""
File: fizzbuzzprime.py V1.0
Auth: Bankole Esan
Desc: Pirple.com Python Homework #5
Task: fizz buzz prime
"""
import time
timeout = .3  # set timeout length
Fizz = "Fizz"  # set Fizz word
Buzz = "Buzz"  # set Buzz word
Prime = "Prime"  # set Prime word


def isPrimeNumber(arg):
    if arg >= 2:
        for i in range(2, arg):
            if arg % i == 0:
                return False
    else:
        return False
    return True


for i in range(1, 101):
    if isPrimeNumber(i):
        print(Prime)
        time.sleep(timeout)
        continue
    if (i % 3 == 0) and (i % 5 == 0):
        print(Fizz + Buzz)
        time.sleep(timeout)
        continue
    if i % 3 == 0:
        print(Fizz)
        time.sleep(timeout)
        continue
    if i % 5 == 0:
        print(Buzz)
        time.sleep(timeout)
        continue
    print(i)
    time.sleep(timeout)

File: pirpleJavascript/fizzbuzzprime.js
/*
 * File: fizzbuzzprime.js V1.0
 * Auth: Bankole Esan
 * Desc: Pirple.com JavaScript Homework #6
 * Task: fizz buzz prime
 */

//  set variables
const fizz = 'Fizz';
const buzz = 'Buzz';
const prime = 'Prime';

const isPrimeNumber = (arg) => {
  if (arg >= 2) {
    for (let i = 2; i < arg; i++) {
      if (arg % i == 0) {
        return false;
      }
    }
  } else {
    return false;
  }
  return true;
};

let i = 1;
do {
  if (isPrimeNumber(i)) {
    console.log(prime);
    i++;
    continue;
  }
  if (i % 3 == 0 && i % 5 == 0) {
    console.log(`${fizz}${buzz}`);
    i++;
    continue;
  }
  if (i % 3 == 0) {
    console.log(fizz);
    i++;
    continue;
  }
  if (i % 5 == 0) {
    console.log(buzz);
    i++;
    continue;
  }
  console.log(i);
  i++;
} while (i < 100);
